go/libraries/doltcore/diff/table_deltas.go

```diff
@@ -517,3 +522,233 @@ func fkSlicesAreEqual(from, to []doltdb.ForeignKey) bool {
+			}
+		}
+	}
+
+	// Print changes between a primary key set change. It contains an ALTER TABLE DROP and an ALTER TABLE ADD
+	if !schema.ColCollsAreEqual(fromSch.GetPKCols(), toSch.GetPKCols()) {
+		ddlStatements = append(ddlStatements, sqlfmt.AlterTableDropPks(td.ToName))
+		if toSch.GetPKCols().Size() > 0 {
+			ddlStatements = append(ddlStatements, sqlfmt.AlterTableAddPrimaryKeys(td.ToName, toSch.GetPKCols()))
+		}
+	}
+
+	for _, idxDiff := range DiffSchIndexes(fromSch, toSch) {
+		switch idxDiff.DiffType {
+		case SchDiffNone:
+		case SchDiffAdded:
+			ddlStatements = append(ddlStatements, sqlfmt.AlterTableAddIndexStmt(td.ToName, idxDiff.To))
+		case SchDiffRemoved:
+			ddlStatements = append(ddlStatements, sqlfmt.AlterTableDropIndexStmt(td.FromName, idxDiff.From))
+		case SchDiffModified:
+			ddlStatements = append(ddlStatements, sqlfmt.AlterTableDropIndexStmt(td.FromName, idxDiff.From))
+			ddlStatements = append(ddlStatements, sqlfmt.AlterTableAddIndexStmt(td.ToName, idxDiff.To))
+		}
+	}
+
+	for _, fkDiff := range DiffForeignKeys(td.FromFks, td.ToFks) {
+		switch fkDiff.DiffType {
+		case SchDiffNone:
+		case SchDiffAdded:
+			parentSch := toSchemas[fkDiff.To.ReferencedTableName]
+			ddlStatements = append(ddlStatements, sqlfmt.AlterTableAddForeignKeyStmt(fkDiff.To, toSch, parentSch))
+		case SchDiffRemoved:
+			ddlStatements = append(ddlStatements, sqlfmt.AlterTableDropForeignKeyStmt(fkDiff.From))
+		case SchDiffModified:
+			ddlStatements = append(ddlStatements, sqlfmt.AlterTableDropForeignKeyStmt(fkDiff.From))
+
+			parentSch := toSchemas[fkDiff.To.ReferencedTableName]
+			ddlStatements = append(ddlStatements, sqlfmt.AlterTableAddForeignKeyStmt(fkDiff.To, toSch, parentSch))
+		}
+	}
+
+	return ddlStatements, nil
+}
+
+func GetDataDiffStatement(tableName string, sch schema.Schema, row sql.Row, rowDiffType ChangeType, colDiffTypes []ChangeType) (string, error) {
+	if len(row) != len(colDiffTypes) {
+		return "", fmt.Errorf("expected the same size for columns and diff types, got %d and %d", len(row), len(colDiffTypes))
+	}
+
+	switch rowDiffType {
+	case Added:
+		return sqlfmt.SqlRowAsInsertStmt(row, tableName, sch)
+	case Removed:
+		return sqlfmt.SqlRowAsDeleteStmt(row, tableName, sch, 0)
+	case ModifiedNew:
+		updatedCols := set.NewEmptyStrSet()
+		for i, diffType := range colDiffTypes {
+			if diffType != None {
+				updatedCols.Add(sch.GetAllCols().GetByIndex(i).Name)
+			}
+		}
+		return sqlfmt.SqlRowAsUpdateStmt(row, tableName, sch, updatedCols)
+	case ModifiedOld:
+		// do nothing, we only issue UPDATE for ModifiedNew
+		return "", nil
+	default:
+		return "", fmt.Errorf("unexpected row diff type: %v", rowDiffType)
+	}
+}
+
+func getCreateTableStmt(tblName string, sch schema.Schema, pkSchema sql.PrimaryKeySchema, fks []doltdb.ForeignKey, fksParentSch map[string]schema.Schema) (string, error) {
+	sqlSch := pkSchema.Schema
+	colStmts := make([]string, len(sqlSch))
+
```

Comment:
Thanks for refactoring out as much as we could to share this logic from one place in GMS. Could you just add a note here in the code to explain to future readers why we can't share more? The goal is to encode some of the context and intent so that future readers understand the direction we want to move in and the constraints we hit that made us stop short of having a single function we could call to get the full create table statement generated. 

I'd probably mention something about us wanting to keep that `create table` statement generation code in one place and mention that because Dolt uses it's own Schema and Column types that aren't directly compatible with GMS' versions, we refactored out common generation parts, but ideally we could find a way to smooth that out even more in the future.